validation/chOpenApiValidator.go

```diff
@@ -30,93 +31,80 @@ var (
 	callOpenApiFilterValidateRequest = openapi3filter.ValidateRequest
 	callGetCHErrors                  = getCHErrors
 	callFindRoute                    = findRoute
+	callOnce	 sync.Once
 )
 
 // CHValidator provides an interface to interact with the CH Validator.
 type CHValidator interface {
 	ValidateRequestAgainstOpenApiSpec(httpReq *http.Request, openApiSpec, contextId string) ([]byte, error)
+	GetSchema(ctx context.Context, openApiSpec, contextId string) error
 }
 
 // CHValidatorImpl is a concrete implementation of the CHValidator interface.
 type CHValidatorImpl struct {
+	doc 	*openapi3.T
 }
 
 // NewCHValidator returns a new CHValidator implementation.
 func NewCHValidator() CHValidator {
-	return CHValidatorImpl{}
+	return &CHValidatorImpl{}
 }
 
 // ValidateRequestAgainstOpenApiSpec takes a request and an openAPI spec location (string relative path) and uses the
 // spec to validate the provided request. If any validation errors are found, then they are formatted and returned to the
 // caller. If any errors are encountered while attempting to validate, they are handled and also returned to the caller.
-func (chv CHValidatorImpl) ValidateRequestAgainstOpenApiSpec(httpReq *http.Request, openApiSpec, contextId string) ([]byte, error) {
+func (chv *CHValidatorImpl) ValidateRequestAgainstOpenApiSpec(httpReq *http.Request, openApiSpec, contextId string) ([]byte, error) {
 
 	// Get the Open API 3 validation schema location.
 	ctx := context.Background()
-	loader := &openapi3.Loader{Context: ctx, IsExternalRefsAllowed: true}
-	abs, err := callFilepathAbs(openApiSpec)
-	if err != nil {
-		log.ErrorC(contextId, err, log.Data{config.MessageKey: "error occurred while retrieving absolute path of validation schema file"})
+
+	if err := chv.GetSchema(ctx, openApiSpec, contextId); err != nil {
```

Comment:
Does this have to be exported, do we use it anywhere outside of the package? (GetSchema)

Reply:
Reworked this logic to make getSchema as non exported function as this is only used in the same package